### DISCUSSION TITLE
umi4 每次热更新速度都要在5秒以上,正常吗?

### DISCUSSION THREAD

**Discussion by lzw10168** [umijs/umi#10617] [General]
mfsu: {  strategy: 'eager' },
  fastRefresh: true,

使用的这俩个配置

**Comment by fz6m** [comment]
这两个选项是默认值，不配置也是这样的。需要给一个热更新慢的最小复现。

**Reply by lzw10168** [reply]
是因为项目太大了吗?  需要做什么能让热更新回到500ms? esbuild吗?

**Reply by lzw10168** [reply]
改一个字母也要5秒的热更新时间, emm. 是我姿势不对嘛

**Reply by lzw10168** [reply]
实验了一下, 删除90%的路由后, 热更新能回到1000ms

**Reply by fz6m** [reply]
目前已知的信息：

1. 尝试关闭 `mfsu: false` 看看热更新时间是否会减少。

2. 尝试手动用 `React.lazy` 分包，分包方式见：https://umijs.org/blog/code-splitting ，特别是对需要加载重依赖的组件部分拆分，比如编辑器等。

3. 使用较多 dva 、model 可能导致扫描项目时间增长，更改刷新较慢。

4. 若没有使用额外的 babel 插件，尝试使用 `srcTranspiler: 'swc'` 提升编译速度（ [srcTranspiler](https://umijs.org/docs/api/config#srctranspiler) ）。

5. 升级 umi 版本到最新。
